### PR TITLE
WIP Add grains exercise.

### PR DESCRIPTION
**Pull request by canweriotnow**
Don't merge yet, still figuring out integer overflow issue.

Sending PR in hopes of suggestions on integer size problem.

### CODE REVIEW

**Comment by wobh** [comment]
http://www.gnu.org/software/emacs/manual/html_mono/calc.html#Calling-Calc-from-Your-Programs ?

File: grains/grains.el [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+;;; Commentary:
+
+;;; Code:
+
+(require 'cl)
+
+(defun square (n)
+  "Find the number of grains on square N."
+  (expt 2 (- n 1)))
```
Comment (wobh) [inline grains/grains.el]: `(calc-eval (format "2^(%d - 1)" n))`

File: grains/grains.el [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+(defun total ()
+  "Find the total number of grains on the board."
+  (reduce #'+ (mapcar #'square (number-sequence 1 64))))
```
Comment (wobh) [inline grains/grains.el]: `(calc-eval "2^64 - 1")`

**Comment by bordeltabernacle** [comment]
Hi @canweriotnow Is this still being worked on, or can this PR be closed?

**Pull request merged**